packages/common-all/src/dnode.ts

```diff
@@ -334,6 +334,14 @@ export class DNodeUtils {
   static getLeafName(note: DNodeProps) {
     return _.split(note.fname, ".").pop();
   }
+
+  static cleanFname(fname: string) {
+    if (_.isUndefined(fname) || fname === "null") return;
+    if (fname.indexOf(" ") >= 0) {
```

Comment:
its not just spaces but also special characters we want to escape. you can use `_.kebabCase` to do the escaping